### PR TITLE
add GALAXY_MEMORY_MB

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

**Review by mira-miracoli — APPROVED** [review]

**Comment by kysrpex** [comment]
As @bernt-matthias pointed out, is this not supposed to be [set by the actual runner](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/util/job_script/MEMORY_STATEMENT_TEMPLATE.sh)? At the moment it only works for SLURM and SGE but imo the solution should then be to add support for HTCondor there too, right?

**Comment by wm75** [comment]
and shouldn't there be a bit of overhead be subtracted from mem if doing the calculation here?

**Comment by mira-miracoli** [comment]
@wm if you mean the overhead the machine needs, I think HTCondor substracts that, when avertising the memory.
eg. a 20G VM advertises only 18834 MB 
I think this happens through this line in the condor_config.local file:
```bash
RESERVED_MEMORY = 1024
```

**Comment by kysrpex** [comment]
> As @bernt-matthias pointed out, is this not supposed to be [set by the actual runner](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/util/job_script/MEMORY_STATEMENT_TEMPLATE.sh)? At the moment it only works for SLURM and SGE but imo the solution should then be to add support for HTCondor there too, right?

I am now wondering though, what is the advantage of letting the runnner set the variable instead of just setting it in the default tool?

**Comment by mira-miracoli** [comment]
I think it is a style choice. Setting it in the runner lets you change it for other runners, like Pulsar runners

**Comment by bernt-matthias** [comment]
> and shouldn't there be a bit of overhead be subtracted from mem if doing the calculation here? 

Interestingly this goes in a similar direction as https://github.com/galaxyproject/galaxy/issues/15952 .. difference is probably that you refer to some machine (HTcondor) overheads and in the issue more like application overheads (not entirely sure yet).

I guess setting `GALAXY_MEMORY_MB` should actually work (at least for non SGE and SLURM systems .. for those I'm not sure which is set first) .. but I guess setting it in the job template would be preferable.

> I am now wondering though, what is the advantage of letting the runnner set the variable instead of just setting it in the default tool?

One advantage might be that all HTcondor based systems would benefit, and not only those using TPV + this very default tool setting. Still for usegalaxy.eu it might be at least a good workaround.

**Comment by kysrpex** [comment]
> One advantage might be that all HTcondor based systems would benefit, and not only those using TPV + this very default tool setting. Still for usegalaxy.eu it might be at least a good workaround.

This is a strong point in favour of setting it in the runner imo.

**Review by kysrpex — APPROVED** [review]
I think it can still take a while until we come up with the best solution for letting the runner set it, and a fix is needed for [rgrnastar](https://github.com/galaxyproject/tools-iuc/tree/main/tools/rgrnastar). I expect no harm from the change so as a workaround I am also ok with it.

File: files/galaxy/tpv/tool_defaults.yml [hunk 1]
```diff
@@ -6,7 +7,8 @@ tools:
     cores: 1
     mem: cores * 3.8
     gpus: 0
-    env: {}
+    env:
+      GALAXY_MEMORY_MB: "{mem * 1024}"
```
Comment (mira-miracoli) [inline files/galaxy/tpv/tool_defaults.yml:11]:
```suggestion
      GALAXY_MEMORY_MB: "{mem * 1024}" # set 5/2023 might be moved to runner or tool wrappers, related to Galaxy issue 15952
```

Comment (kysrpex) [reply files/galaxy/tpv/tool_defaults.yml:11]: Alternatively, making this part of the commit message may be equally useful.

**Pull request merged**